xtable-core/src/main/java/org/apache/xtable/avro/AvroSchemaConverter.java

```diff
@@ -435,6 +435,11 @@ private Schema fromInternalSchema(InternalSchema internalSchema, String currentP
             Schema.createFixed(
                 internalSchema.getName(), internalSchema.getComment(), null, fixedSize),
             internalSchema);
+      case UUID:
+        // Convert to fixed 16-byte array since Avro does not support UUID natively
+        return finalizeSchema(
+            Schema.createFixed(internalSchema.getName(), internalSchema.getComment(), null, 16),
```

Comment:
`Schema` exposes an `addProp` method. We could try to add a prop here that allows us to determine that a fixed byte array should be treated as an UUID in XTable

Reply:
> technically there is a logical type for UUID in recent versions of avro https://avro.apache.org/docs/1.10.0/spec.html#UUID, but this is mentioning it is really just an annotation on a string field.

> `Schema` exposes an `addProp` method. We could try to add a prop here that allows us to determine that a fixed byte array should be treated as an UUID in XTable

Already added corresponding Prop in latest commit, thanks for providing this context Tim!